modules/virt-creating-a-service-from-a-virtual-machine.adoc

```diff
@@ -0,0 +1,196 @@
+service.
+
+This procedure presents an example of how to create, connect to, and expose a Service object
+of `type: ClusterIP` as a virtual machine-backed service.
+
+[NOTE]
+====
+ClusterIP is the default service `type`, if the service `type` is not specified.
+====
+
+.Procedure
+
+. Edit the `VirtualMachine` YAML configuration file.
+. Add the label `special: key` in the `metadata` section.
++
+
+[NOTE]
+====
+Labels on a `VirtualMachine` are passed through to the pod. The labels on
+the `VirtualMachine`, for example `special: key`, must match the labels in
+the Service  YAML `selector` attribute, which you create later
+in this procedure.
+====
++
+
+[source,yaml]
+----
+aapiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  name: vm-ephemeral
+  namespace: example-namespace
+spec:
+  running: false
+  template:
+    metadata:
+      labels:
+        special: key <1>
+    spec:
+      domain:
+        devices:
+          disks:
+            - name: containerdisk
+              disk:
+                bus: virtio
+            - name: cloudinitdisk
+              disk:
+                bus: virtio
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 1024M
+      networks:
+        - name: default
+          pod: {}
+      volumes:
+        - name: containerdisk
+          containerDisk:
+            image: kubevirt/fedora-cloud-container-disk-demo
+        - name: cloudinitdisk
+          cloudInitNoCloud:
+            userData: |
+              #!/bin/bash
+              echo "fedora" | passwd fedora --stdin
+----
+<1> The `special: key` label
++
+
+. Edit the `Service` YAML to create and expose the Service object, connecting by
+specifying the service name. `Service` is scoped by `namespace`.
+.. Specify the `name` for the service you are creating and exposing.
+.. Add a `namespace` key to the `metadata` section of the Service YAML, that corresponds to the
+`namespace` you referenced in the virtual machine YAML.
+.. In the `spec` section of the Service YAML, add `special: key` to the `selector` attribute,
+which corresponds to the `labels` you added in the virtual machine YAML configuration file.
+.. In the `spec` section of the Service YAML, add `type: ClusterIP` for a
+ClusterIP service. Add `targetPort: 22`, exposing the service on SSH port `22`.
++
+To create and expose other types of services externally, outside of the cluster,
+such as `NodePort` and `LoadBalancer`, replace
+`type: ClusterIP` with `type: NodePort` or `type: LoadBalancer`, as appropriate,
+in the Service object definition.
++
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: vmservice <1>
+  namespace: example-namespace <2>
+spec:
+  ports:
+  - port: 27017
+    protocol: TCP
+    targetPort: 22 <3>
+  selector:
+    special: key <4>
+  type: ClusterIP <5>
+----
+<1> Service name
```

Comment:
These callouts should be consistent with periods and whether they are sentences or not. Also I would try to avoid starting with a lowercase like `namespace` and `selector` below.

But if you take the suggestion to move the steps above into the callouts, then that should take care of this.

Reply:
Moved steps into callouts as complete imperative sentences. Great suggestion.